### PR TITLE
Fixes #17950: testing loop should report an error when rudderc does

### PR DESCRIPTION
**Pull request by gpoblon**
https://issues.rudder.io/issues/17950

### CODE REVIEW

File: rudder-lang/tools/tester.sh [hunk 1]
```diff
@@ -106,16 +122,15 @@ fi
...
-  ${cfjson_tester} ncf-to-json --config-file=${config_file} "${technique_path}" "${technique}.json" \
-  && ${rudderc} --config-file=${config_file} --translate -s "${technique}.json" \
-  && ${rudderc} --config-file=${config_file} -s "${technique}.rl" \
-  && ${cfjson_tester} ncf-to-json --config-file="${config_file}" "${technique}.rl.cf" "${technique}.rl.cf.json" \
-  && ${cfjson_tester} compare-json --config-file="${config_file}" "${technique}.json" "${technique}.rl.cf.json" \
-  && ${cfjson_tester} compare-cf --config-file="${config_file}" "${technique_path}" "${technique}.rl.cf"
+  ${cfjson_tester} ncf-to-json --config-file=${config_file} "${technique_path}" "${technique}.json" && status=${status}+1 \
+    && ${rudderc} --config-file=${config_file} --translate -s "${technique}.json" && status=${status}+1 \
+    && ${rudderc} --config-file=${config_file} -s "${technique}.rl" && status=${status}+1 \
+    && ${cfjson_tester} ncf-to-json --config-file="${config_file}" "${technique}.rl.cf" "${technique}.rl.cf.json" && status=${status}+1 \
```
Comment (peckpeck) [inline rudder-lang/tools/tester.sh:131]: status=${status}+1 it not proper shell you must write 
status=$((${status}+1)) or status=$((status+1))

Comment (gpoblon) [reply rudder-lang/tools/tester.sh:131]: fixed

**Comment by gpoblon** [comment]
PR updated with a new commit

**Comment by gpoblon** [comment]
OK, squash merging this PR

**Pull request merged**